### ISSUE TITLE
Cayley-Hamilton theorem

### ISSUE THREAD

**Issue by Roland-Sands** [sympy/sympy#14266]
Would it be worthwhile implementing the Cayley-Hamilton theorem, would it help in calculating powers of matrices which cannot be diagonalized

**Comment by sidhantnagpal** [comment]
Yes, actually not only can it be help for calculating powers of matrices which cannot be diagonalized, it will also be used as an intermediary for evaluation of Homogeneous Linear Recurrences (of higher order) with Constant Coefficients which I have proposed on the SymPy Google Group.

**Comment by Roland-Sands** [comment]
> I would like to take this up

Excellent idea
